www/templates/jsontestresults.html

```diff
@@ -3,49 +3,62 @@
 {% block suites %}
    <div class="content container">
     {% for suite in data.suites %}
+        <div class="table-holder">
+            {% set additionalCommandLineArguments = '' %}
+            {% if suite.minimalCommandLine %}
+                {% set additionalCommandLineArguments = suite.minimalCommandLine|join(" ") %}
+            {% endif %}
 
-        {% set additionalCommandLineArguments = '' %}
-        {% if suite.minimalCommandLine %}
-            {% set additionalCommandLineArguments = suite.minimalCommandLine|join(" ") %}
-        {% endif %}
+            {% if suite.summary.result %}
+                <div class="row">
+                    <h4>Suite type: {{ suite.name }}
+                         {% if suite.description %}
+                            {{ suite.description }}
+                         {% endif %}
+                    </h4>
+                    {% if data.utrPrefix %}
+                        <a class="label label-default collapse" data-target="#suiteCommand{{ loop.index }}">Command to run locally</a>
 
-        {% if suite.summary.result %}
-            <div class="row">
-                <h4>Suite type: {{ suite.name }}
-                     {% if suite.description %}
-                        {{ suite.description }}
-                     {% endif %}
-                </h4>
-                {% if suite.details %}
-                    <h4>{{ suite.details }}</h4>
+                        <div id="suiteCommand{{ loop.index }}" class="report-command">
+                            <input class="report-command-text" type="text" value="{{ data.utrPrefix }} --suite={{ suite.name }} {{ additionalCommandLineArguments }}" onclick="this.select();" readonly/>
+                        </div>
+                    {% endif %}
+                </div>
+
+                {% if suite.error %}
+                    <div class="failure">{{ suite.error }}</div>
                 {% endif %}
-                {% if data.utrPrefix %}
-                    <a class="label label-default collapse" data-target="#suiteCommand{{ loop.index }}">Command to run locally</a>
 
-                    <div id="suiteCommand{{ loop.index }}" class="report-command">
-                        <input class="report-command-text" type="text" value="{{ data.utrPrefix }} --suite={{ suite.name }} {{ additionalCommandLineArguments }}" onclick="this.select();" readonly/>
-                    </div>
+                {% if suite.failureReasons %}
```

Comment:
minor: This `if` and `div` below is not mandatory. If you have an empty list, the for loop doesn't execute a command inside, so it won't display any div with fail.

Reply:
it might be null in old version

Reply:
not 100% sure